velox/functions/prestosql/window/tests/NthValueTest.cpp

```diff
@@ -233,34 +251,157 @@ TEST_F(NthValueTest, multiInputRangeFrames) {
       {vectors}, "nth_value(c0, 5)", kFrameOverClauses, kRangeFrameClauses);
 }
 
-TEST_F(NthValueTest, basicRowFrames) {
+TEST_F(NthValueTest, basicUnboundedRowFrames) {
   auto vectors = makeBasicVectors(50);
 
   testWindowFunction(
-      {vectors}, "nth_value(c0, c2)", kFrameOverClauses, kRowsFrameClauses);
+      {vectors},
+      "nth_value(c0, c2)",
+      kFrameOverClauses,
+      kUnboundedRowsFrameClauses);
   testWindowFunction(
-      {vectors}, "nth_value(c0, 1)", kFrameOverClauses, kRowsFrameClauses);
+      {vectors},
+      "nth_value(c0, 1)",
+      kFrameOverClauses,
+      kUnboundedRowsFrameClauses);
   testWindowFunction(
-      {vectors}, "nth_value(c0, 5)", kFrameOverClauses, kRowsFrameClauses);
+      {vectors},
+      "nth_value(c0, 5)",
+      kFrameOverClauses,
+      kUnboundedRowsFrameClauses);
 }
 
-TEST_F(NthValueTest, singlePartitionRowFrames) {
+TEST_F(NthValueTest, singlePartitionUnboundedRowFrames) {
   auto vectors = makeSinglePartitionVectors(400);
 
   testWindowFunction(
-      {vectors}, "nth_value(c0, c2)", kFrameOverClauses, kRowsFrameClauses);
+      {vectors},
+      "nth_value(c0, c2)",
+      kFrameOverClauses,
+      kUnboundedRowsFrameClauses);
   testWindowFunction(
-      {vectors}, "nth_value(c0, 5)", kFrameOverClauses, kRowsFrameClauses);
+      {vectors},
+      "nth_value(c0, 5)",
+      kFrameOverClauses,
+      kUnboundedRowsFrameClauses);
 }
 
-TEST_F(NthValueTest, multiInputRowFrames) {
+TEST_F(NthValueTest, multiInputUnboundedRowFrames) {
   auto vectors = makeSinglePartitionVectors(200);
   auto doubleVectors = {vectors, vectors};
 
   testWindowFunction(
-      doubleVectors, "nth_value(c0, c2)", kFrameOverClauses, kRowsFrameClauses);
+      doubleVectors,
+      "nth_value(c0, c2)",
+      kFrameOverClauses,
+      kUnboundedRowsFrameClauses);
```

Comment:
Consider changing this test to a parameterized test with frame clauses used as a parameter to avoid repetition.

Reply:
Could you please confirm if all the new frame testcases added should be removed and the existing tests be parameterized on frameClause instead? Thanks!

Reply:
Got it, will update accordingly, thanks!